src/main/scala/net/ripe/rpki/nro/service/Ports.scala

```diff
@@ -212,9 +230,30 @@ object Ports extends Logging {
     }
   }
 
-  def readConflicts(conflictFile: String = s"${config.previousConflictFile}"): List[Conflict] = {
-    logger.debug(s"Reading conflicts from ${config.previousConflictFile}")
-    Using.resource(CSVReader.open(new File(conflictFile))){ reader =>
+  private def fetchAndWriteLocally(source: String, dest: String): Unit = {
+    val fetchResponse: Future[Response] = fetchWithRetries(source)
+    Try(Await.result(fetchResponse, Duration.Inf)) match {
+      case Success(response) if response.statusCode == 200 =>
+        val responseText = response.text()
+        Using.resource(new PrintWriter(new File(dest))) { writer =>
+          writer.write(responseText)
+          logger.info(s"---Done fetching $source into $dest---\n\n\n")
+        }
+      case _ =>
+        logger.error(s"Failed to fetch $source after $maxRetries retries")
+        System.exit(1)
+    }
+  }
+  def readConflicts(conflictURL: String ): List[Conflict] = {
+    logger.debug(s"Reading conflicts from $conflictURL" )
+
+    val localConflictCSV = if (conflictURL.startsWith("http")) {
+      val localConflict = conflictURL.replaceAll("\\W", "");
```

Comment:
Using a parsed URL helps against these cases. Also I would rather try to avoid having `file://` URLs in config, or worse using other schemas. We always need to fetch the files anyway, and only support HTTP fetches (and `https` obviously) cleans up the code.

Reply:
Agreed, adjusted the code so it is now expecting to fetch http/s, rejecting early on cli when it fails to parse URL.